geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -215,49 +213,39 @@ object AccumuloDataStoreFactory {
-  val instanceIdParam        = new Param("instanceId", classOf[String], "Accumulo Instance ID", true)
-  val zookeepersParam        = new Param("zookeepers", classOf[String], "Zookeepers", true)
-  val userParam              = new Param("user", classOf[String], "Accumulo user", true)
-  val passwordParam          = new Param("password", classOf[String], "Accumulo password", false, null, Collections.singletonMap(Parameter.IS_PASSWORD, java.lang.Boolean.TRUE))
-  val keytabPathParam        = new Param("keytabPath", classOf[String], "Path to keytab file", false)
-  val authsParam             = org.locationtech.geomesa.security.AuthsParam
-  val visibilityParam        = new Param("visibilities", classOf[String], "Default Accumulo visibilities to apply to all written data", false)
-  val tableNameParam         = new Param("tableName", classOf[String], "Accumulo catalog table name", true)
-  val queryTimeoutParam      = GeoMesaDataStoreFactory.QueryTimeoutParam
-  val queryThreadsParam      = GeoMesaDataStoreFactory.QueryThreadsParam
-  val recordThreadsParam     = new Param("recordThreads", classOf[Integer], "The number of threads to use for record retrieval", false, 10)
-  val writeThreadsParam      = new Param("writeThreads", classOf[Integer], "The number of threads to use for writing records", false, 10)
-  val looseBBoxParam         = GeoMesaDataStoreFactory.LooseBBoxParam
-  val generateStatsParam     = GeoMesaDataStoreFactory.GenerateStatsParam
-  val collectQueryStatsParam = new Param("collectQueryStats", classOf[java.lang.Boolean], "Collect statistics on queries being run", false, true)
-  val auditQueriesParam      = GeoMesaDataStoreFactory.AuditQueriesParam
-  val cachingParam           = GeoMesaDataStoreFactory.CachingParam
-  val mockParam              = new Param("useMock", classOf[String], "Use a mock connection (for testing)", false)
-  val forceEmptyAuthsParam   = org.locationtech.geomesa.security.ForceEmptyAuthsParam
-  val NamespaceParam         = new Param("namespace", classOf[String], "Namespace", false)
+  val ConnectorParam       = new GeoMesaParam[Connector]("accumulo.connector", "Accumulo connector", deprecated = Seq("connector"))
+  val InstanceIdParam      = new GeoMesaParam[String]("accumulo.instance.id", "Accumulo Instance ID", required = true, deprecated = Seq("instanceId", "accumulo.instanceId"))
+  val ZookeepersParam      = new GeoMesaParam[String]("accumulo.zookeepers", "Zookeepers", required = true, deprecated = Seq("zookeepers"))
+  val UserParam            = new GeoMesaParam[String]("accumulo.user", "Accumulo user", required = true, deprecated = Seq("user"))
+  val PasswordParam        = new GeoMesaParam[String]("accumulo.password", "Accumulo password", metadata = Map(Parameter.IS_PASSWORD -> java.lang.Boolean.TRUE), deprecated = Seq("password"))
+  val KeytabPathParam      = new GeoMesaParam[String]("accumulo.keytab.path", "Path to keytab file", deprecated = Seq("keytabPath", "accumulo.keytabPath"))
```

Comment:
Is the goal of the deprecated Seq to add both the old key and a hastily namespaced one?

That seems like a damn nice call:).

Reply:
yeah the namespaced one was used by the lambda ds